### DISCUSSION TITLE
How to make tree-shaking work?

### DISCUSSION THREAD

**Discussion by R-Oscar** [floating-ui/floating-ui#2629] [Q&A]
I've made a setup with webpack + babel + floating-ui

Here is a webpack config:
```
const path = require('path');

module.exports = {
  mode: "production",
  entry: './app.js',
  output: {
    filename: 'main.js',
    path: path.resolve(__dirname, 'dist'),
  },
  module: {
    rules: [
      {
        test: /\.(?:js|mjs|cjs|ts)$/,
        exclude: /node_modules/,
        use: {
          loader: 'babel-loader',
          options: {
            "presets": [
              [
                "@babel/preset-env",
                {
                  "useBuiltIns": "entry",
                  "corejs": "3.22",
                  "modules": false
                }
              ]
            ]
          }
        }
      }
    ]
  },
  optimization: {
    minimize: false,
    usedExports: true
  }
};
```

And an entrypoint app.js file:
```
import { flip } from '@floating-ui/core';

console.log(flip, 'test');
```

As you can see, I'm only importing `flip` middleware from core package. However,  seems like there's a complete library with functions such as `computePosition`, `shift`, and other middlewares in my bundle. Why is this happening? I had tried to do the same thing with `lodash-es` before, and it has bundled the only function that I used (so tree-shaking worked).

**Comment by atomiks** [comment]
Use Vite instead 🤷 ? It should work by default

**Reply by R-Oscar** [reply]
It indeed works in Vite, though I can't use it due to webpack being used everywhere in our company. Any hints on how to fix it, or where to look for? Or you're implying that this is a webpack issue?

**Reply by FezVrasta** [reply]
It is a webpack configuration issue, if it was an issue on our side it would not work on any bundler at all.
